### PR TITLE
LPD-34377 - WebURL api (part2)

### PR DESCRIPTION
**Pull request by brbalazs**
https://liferay.atlassian.net/browse/LPD-34377

cc @smottal

### CODE REVIEW

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java [hunk 1]
```diff
@@ -36,6 +42,28 @@
...
+	public void deleteWebUrlByExternalReferenceCode(
+			String externalReferenceCode)
+		throws Exception {
+
+		Website website = _websiteService.fetchWebsiteByExternalReferenceCode(
+			externalReferenceCode, contextCompany.getCompanyId());
+
+		if (website == null) {
+			throw new NoSuchWebsiteException(
+				"Unable to get Website with external reference code " +
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java:60]: Unable to find website with...

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java [hunk 2]
```diff
@@ -115,12 +143,77 @@ public WebUrl getWebUrl(Long webUrlId) throws Exception {
...
+	@Override
+	public WebUrl getWebUrlByExternalReferenceCode(String externalReferenceCode)
+		throws Exception {
+
+		Website website = _websiteService.fetchWebsiteByExternalReferenceCode(
+			externalReferenceCode, contextCompany.getCompanyId());
+
+		if (website == null) {
+			throw new NoSuchWebsiteException(
+				"Unable to get Website with external reference code " +
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java:155]: Unable to find website with...

File: modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java [hunk 3]
```diff
@@ -115,12 +143,77 @@ public WebUrl getWebUrl(Long webUrlId) throws Exception {
...
+	public WebUrl patchWebUrlByExternalReferenceCode(
+			String externalReferenceCode, WebUrl webUrl)
+		throws Exception {
+
+		Website website = _websiteService.fetchWebsiteByExternalReferenceCode(
+			externalReferenceCode, contextCompany.getCompanyId());
+
+		if (website == null) {
+			throw new NoSuchWebsiteException(
+				"Unable to get Website with external reference code " +
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/resource/v1_0/WebUrlResourceImpl.java:189]: Unable to find website with...

**Review by smottal — CHANGES_REQUESTED** [review]

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:sf

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java [hunk 4]
```diff
@@ -75,10 +75,23 @@ protected WebUrl randomWebUrl() {
...
+	@Override
+	protected WebUrl testDeleteWebUrl_addWebUrl() throws Exception {
+		return _addWebUrl();
+	}
+
+	@Override
+	protected WebUrl testDeleteWebUrlByExternalReferenceCode_addWebUrl()
+		throws Exception {
+
+		return testDeleteWebUrl_addWebUrl();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java:92]: return _addWebUrl();

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java [hunk 5]
```diff
@@ -185,16 +198,39 @@ protected Long testGetUserAccountWebUrlsPage_getUserAccountId() {
...
-			randomWebUrl(), Contact.class.getName(), _user.getContactId(),
-			ListTypeConstants.CONTACT_WEBSITE);
+		return _addWebUrl();
+	}
+
+	@Override
+	protected WebUrl testGetWebUrlByExternalReferenceCode_addWebUrl()
+		throws Exception {
+
+		return testGetWebUrl_addWebUrl();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java:208]: return _addWebUrl();

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java [hunk 6]
```diff
@@ -185,16 +198,39 @@ protected Long testGetUserAccountWebUrlsPage_getUserAccountId() {
...
+	@Override
+	protected WebUrl testPatchWebUrl_addWebUrl() throws Exception {
+		return _addWebUrl();
+	}
+
+	@Override
+	protected WebUrl testPatchWebUrlByExternalReferenceCode_addWebUrl()
+		throws Exception {
+
+		return testPatchWebUrl_addWebUrl();
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java:225]: return _addWebUrl();

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java [hunk 7]
```diff
@@ -185,16 +198,39 @@ protected Long testGetUserAccountWebUrlsPage_getUserAccountId() {
...
+	}
+
+	private WebUrl _addWebUrl() throws Exception {
+		return _addWebUrl(
+			randomWebUrl(), Contact.class.getName(), _user.getContactId(),
+			ListTypeConstants.CONTACT_WEBSITE);
+	}
+
```
Comment (smottal) [inline modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java:233]: why do we need this? Can we put this 
```
return _addWebUrl(
			randomWebUrl(), Contact.class.getName(), _user.getContactId(),
			ListTypeConstants.CONTACT_WEBSITE);
```
directly where we need it?

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java (hunk 7 above)
Comment (brbalazs) [reply modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/WebUrlResourceTest.java:233]: @smottal sure we can, I just thought that this is easier then repeating the same code a couple of times.

**Comment by brbalazs** [comment]
ci:test:sf

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by smottal** [comment]
ci:test:stable

**Comment by brbalazs** [comment]
ci:stop

**Comment by brbalazs** [comment]
ci:test:relevant

**Comment by brbalazs** [comment]
Now I managed to re-run baseline and updated the PR with semantic versioning commit again. That's why I restarted the tests.

**Comment by smottal** [comment]
ci:test:sf

**Comment by smottal** [comment]
ci:report:28326087

**Comment by smottal** [comment]
ci:test:sf

**Pull request closed without merging**